Shell/FOOLElimination.cpp

```diff
@@ -496,14 +539,16 @@ void FOOLElimination::process(Term* term, Context context, TermList& termResult,
         process(*term->nthArgument(1), context, elseBranch, elseBranchFormula);
 
         // the sort of the term is the sort of the then branch
-        unsigned resultSort = 0;
+        TermList resultSort;
         if (context == TERM_CONTEXT) {
           resultSort = SortHelper::getResultSort(thenBranch, _varSorts);
           ASS_EQ(resultSort, SortHelper::getResultSort(elseBranch, _varSorts));
         }
 
```

Comment:
This used to be initialized to = 0 (default sort), whenever (context != TERM_CONTEXT);
now we leave unitialized `resultSort` in that case. Is it OK to do so for some reason?

Reply:
I think the old version was defensive programming since `SortHelper::getResultSort(thenBranch, _varSorts);` should return a result in all cases. 

I have reintroduced the defense and the line is now:

`TermList resultSort = Term::defaultSort();`